antelope_contracts/tests/erc20/integrated_tests.cpp

```diff
@@ -0,0 +1,482 @@
+    bal = get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount();
+
+    BOOST_REQUIRE(99990010 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount());
+}
+FC_LOG_AND_RETHROW()
+
+
+
+BOOST_FIXTURE_TEST_CASE(it_eos_to_evm, it_tester)
+try {
+    evm_eoa evm1;
+    auto addr_alice = silkworm::make_reserved_address("alice"_n.to_uint64_t());
+    // Give evm1 some EOS
+    transfer_token(eos_token_account, "alice"_n, evm_account, make_asset(1000000, eos_token_symbol), evm1.address_0x().c_str());
+    produce_block();
+
+
+    // USDT balance should be zero
+    auto bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 0);
+    produce_block();
+
+    transfer_token(token_account, "alice"_n, evmtok_account, make_asset(10000, token_symbol), evm1.address_0x().c_str());
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 990000); // +1000000 - 10000
+    BOOST_REQUIRE(99990000 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount()); // -10000
+    produce_block();
+
+    BOOST_REQUIRE_EXCEPTION(transfer_token(token_account, "alice"_n, evmtok_account, make_asset(0, token_symbol), evm1.address_0x().c_str()),
+                            eosio_assert_message_exception, eosio_assert_message_is("must transfer positive quantity"));
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 990000);
+    BOOST_REQUIRE(99990000 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount());
+    produce_block();
+
+
+    BOOST_REQUIRE_EXCEPTION(transfer_token(token_account, "alice"_n, evmtok_account, make_asset(10, token_symbol), evm1.address_0x().c_str()),
+                            eosio_assert_message_exception, eosio_assert_message_is("deposit amount must be greater than ingress fee"));
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 990000);
+    BOOST_REQUIRE(99990000 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount());
+    produce_block();
+
+    BOOST_REQUIRE_EXCEPTION(transfer_token(token_account, "alice"_n, evmtok_account, make_asset(100, token_symbol), evm1.address_0x().c_str()),
+                            eosio_assert_message_exception, eosio_assert_message_is("deposit amount must be greater than ingress fee"));
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 990000);
+    BOOST_REQUIRE(99990000 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount());
+    produce_block();
+
+    transfer_token(token_account, "alice"_n, evmtok_account, make_asset(101, token_symbol), evm1.address_0x().c_str());
+    bal = balanceOf(evm1.address_0x().c_str());
+    BOOST_REQUIRE(bal == 990100); // 9900000 + (10100 - 10000)
+    BOOST_REQUIRE(99989899 == get_balance("alice"_n, token_account, symbol::from_string("4,USDT")).get_amount()); // 99990000 - 101
+    produce_block();
+
```

Comment:
Also check the tokens table to see that `balance` and `fee_balance` have the appropriate values.

Reply:
done